test-src/manageamiset_test.js

```diff
@@ -85,6 +87,11 @@ describe('AMI Set api', () => {
     });
   });
 
+  it('should be able to check if the AMIs from an AMI set are valid', async () => {
+    await client.createAmiSet(id, amiSetDefinition);
+    assume(await client.validateAmiSet(id)).to.deeply.equal([]);
```

Comment:
What I usually end up doing is something like this: https://github.com/taskcluster/aws-provisioner/blob/master/test-src/badworkertype_test.js#L69

There's a slight issue in the implementation of this test, it shouldn't be throwing in the try block, but do you get the idea?  basically we handle the error and if the error is one that we don't expect, we re-throw to mark as a failure.